File: backend/src/main/java/com/facilite/backend/service/ProjetoService.java
package com.facilite.backend.service;



import com.facilite.backend.dto.ProjetoRequest;
import com.facilite.backend.dto.ProjetoResponse;
import com.facilite.backend.model.Projeto;
import com.facilite.backend.model.StatusProjeto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface ProjetoService {

    List<ProjetoResponse> listarTodos();
    ProjetoResponse buscarPorId(Long id);
    ProjetoResponse criarProjeto(ProjetoRequest request);
    ProjetoResponse atualizarProjeto(Long id, ProjetoRequest request);
    void excluirProjeto(Long id);
    List<ProjetoResponse> listarPorStatus(StatusProjeto status);
    ProjetoResponse transicionarStatus(Long id, StatusProjeto novoStatus);
    void calcularEAtualizarMetricas(Projeto projeto);
    Page<ProjetoResponse> listarPaginado(Pageable pageable);
}

File: frontend/src/services/projetoService.ts
import { api } from './api';
import type {ProjetoRequest } from '../types/projeto/projetoRequest';
import type {ProjetoResponse} from '../types/projeto/projetoResponse';
import type { StatusProjeto } from 'src/types/projeto/statusprojeto';

export const projetoService = {
  // CRUD Básico
  listarTodos: (): Promise<ProjetoResponse[]> => 
    api.get('/projetos').then(response => response.data),

  buscarPorId: (id: number): Promise<ProjetoResponse> => 
    api.get(`/projetos/${id}`).then(response => response.data),

  criar: (projeto: ProjetoRequest): Promise<ProjetoResponse> => 
    api.post('/projetos', projeto).then(response => response.data),

  atualizar: (id: number, projeto: ProjetoRequest): Promise<ProjetoResponse> => 
    api.put(`/projetos/${id}`, projeto).then(response => response.data),

  excluir: (id: number): Promise<void> => 
    api.delete(`/projetos/${id}`),

  // Kanban e Status
  listarPaginado: (page: number = 0, size: number = 10): Promise<{ content: ProjetoResponse[] }> => 
    api.get(`/projetos/paginado?page=${page}&size=${size}`).then(response => response.data),

  transicionarStatus: (id: number, novoStatus: StatusProjeto): Promise<ProjetoResponse> => 
    api.patch(`/projetos/${id}/status/${novoStatus}`).then(response => response.data),
};

